contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,491 @@
+        if (isInvalidDeposit(amount_, recipient_)) revert YieldDirector_InvalidDeposit();
+
+        IERC20(sOHM).safeTransferFrom(msg.sender, address(this), amount_);
+        IERC20(sOHM).approve(address(staking), amount_);
+        uint256 gohmAmount = staking.wrap(address(this), amount_);
+
+        depositId = _createDeposit(gohmAmount, recipient_);
+    }
+
+    /**
+        @notice Deposit additional gOHM, and update deposit record
+        @param depositId_ Deposit ID to direct additional gOHM to
+        @param amount_ Amount of new gOHM debt issued from donor to recipient
+    */
+    function addToDeposit(uint256 depositId_, uint256 amount_) external {
+        if (isInvalidUpdate(depositId_, amount_)) revert YieldDirector_InvalidUpdate();
+        if (depositInfo[depositId_].depositor != msg.sender) revert YieldDirector_NotYourDeposit();
+
+        IERC20(gOHM).safeTransferFrom(msg.sender, address(this), amount_);
+
+        _increaseDeposit(depositId_, amount_);
+    }
+
+    /**
+        @notice Deposit additional sOHM, wrap to gOHM, and update deposit record
+        @param depositId_ Deposit ID to direct additional gOHM to
+        @param amount_ Amount of new sOHM debt issued from donor to recipient
+    */
+    function addToSohmDeposit(uint256 depositId_, uint256 amount_) external {
+        if (isInvalidUpdate(depositId_, amount_)) revert YieldDirector_InvalidUpdate();
+        if (depositInfo[depositId_].depositor != msg.sender) revert YieldDirector_NotYourDeposit();
+
+        IERC20(sOHM).safeTransferFrom(msg.sender, address(this), amount_);
+        IERC20(sOHM).approve(address(staking), amount_);
+        uint256 gohmAmount = staking.wrap(address(this), amount_);
+
+        _increaseDeposit(depositId_, gohmAmount);
+    }
+
+    /**
+        @notice Withdraw donor's gOHM from vault
+        @param depositId_ Deposit ID to remove gOHM deposit from
+        @param amount_ Amount of gOHM deposit to remove and return to donor
+    */
+    function withdrawPrincipal(uint256 depositId_, uint256 amount_) external {
+        uint256 amountWithdrawn = _withdraw(depositId_, amount_);
+
+        IERC20(gOHM).safeTransfer(msg.sender, amountWithdrawn);
+    }
+
+    /**
+        @notice Withdraw donor's gOHM from vault, and return it as sOHM
+        @param depositId_ Deposit ID to remove gOHM debt from
+        @param amount_ Amount of gOHM debt to remove and return to donor as sOHM
+    */
+    function withdrawPrincipalAsSohm(uint256 depositId_, uint256 amount_) external {
+        uint256 amountWithdrawn = _withdraw(depositId_, amount_);
+
+        IERC20(sOHM).approve(address(staking), amountWithdrawn);
+        staking.unwrap(msg.sender, amountWithdrawn);
+    }
+
+    /**
+        @notice Withdraw all gOHM from all donor positions
+    */
+    function withdrawAll() external {
+        if (withdrawDisabled) revert YieldDirector_WithdrawalsDisabled();
+
+        uint256[] storage depositIds = depositorIds[msg.sender];
```

Comment:
since you are not modifying depositIds it can be memory. I think it might be more gas efficient to use memory here but you will have to test it. It takes more gas to instantiate the array in memory but if you read from it frequently enough it will overall save more gas than storage.